src/node/tendermint_internals.mli

```diff
@@ -0,0 +1,94 @@
+open Crypto
+
+type height = int64 [@@deriving yojson]
+
+type round = int [@@deriving yojson]
+(** At a specific (chain) height, Tendermint's consensus algorithm may run several rounds.*)
+
+(** Tendermint sometimes decides on a `Nil` value. *)
+type value =
+  | Block of Protocol.Block.t
+  | Nil
+[@@deriving yojson]
+
+val repr_of_value : value -> value
+(* TODO: ConsensusStep2 Tendermint uses repr of values instead of values. *)
+
+(* FIXME: ConsensusStep2 this is copied bad design. *)
+val produce_value : (State.t -> value) ref
+
+val is_valid : State.t -> value -> bool
+(** Ensures a value going through Tendermint is valid. TODO: ConsensusStep2 actually verify signatures, logging system. *)
+
+val on_block : nil_default:'a -> (Protocol.Block.t -> 'a) -> value -> 'a
+
+val block : Protocol.Block.t -> value
+
+val nil : value
+
+val update_value : value -> round -> value
```

Comment:
Hmmmm. Maybe `value -> round -> value result`? May change my mind when I better understand it.